pkg/cmd/dedicated/register/registercluster.go

```diff
@@ -117,23 +115,29 @@ func runRegisterClusterCmd(opts *options) (err error) {
 
 }
 
-func getPaginatedClusterList(opts *options) error {
-	cl, err := clustermgmt.GetClusterList(opts.f, opts.accessToken, opts.clusterManagementApiUrl, opts.pageNumber, opts.pageSize)
+func getSelectedClusterFromId(opts *options, clusterId string) error {
+	cluster, err := clustermgmt.GetClusterById(opts.f, opts.accessToken, opts.clusterManagementApiUrl, clusterId)
 	if err != nil {
 		return err
 	}
-
-	opts.clusterList = validateClusters(cl, opts.clusterList)
+	opts.selectedCluster = *cluster
 	return nil
 }
 
-func validateClusters(clusters *clustersmgmtv1.ClusterList, cls []clustersmgmtv1.Cluster) []clustersmgmtv1.Cluster {
-	for _, cluster := range clusters.Slice() {
-		if cluster.State() == clusterReadyState && cluster.MultiAZ() == true {
-			cls = append(cls, *cluster)
-		}
+func validClusterString() string {
+	return "multi_az='true' AND state='ready'"
+}
+
+func getPaginatedClusterList(opts *options) error {
+	cl, err := clustermgmt.GetClusterListWithSearchParams(opts.f, opts.clusterManagementApiUrl, opts.accessToken, validClusterString(), opts.pageNumber, opts.pageSize)
+	if err != nil {
+		return err
 	}
-	return cls
+
+	for _, cluster := range cl.Slice() {
+		opts.clusterList = append(opts.clusterList, *cluster)
```

Comment:
What happens if this function gets called twice?
Should we avoid possible duplicates?